### ISSUE TITLE
Improve formatting of nested JoinTable

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14311]
The following integration test query from https://github.com/jOOQ/jOOQ/issues/14308 and https://github.com/jOOQ/jOOQ/issues/14310:

```sql
Table<Record> t1 = TBookToBookStore()
    .join(TBookStore())
    .on(TBookToBookStore_BOOK_STORE_NAME().eq(TBookStore_NAME())).as("t1");

Table<Record> t2 = TBook().leftJoin(t1).on(TBook_ID().eq(t1.field(TBookToBookStore_BOOK_ID()))).as("t2");

Result<?> r2 =
create()
    .select(t2.asterisk())
    .from(t2)
    .orderBy(
        t2.field(TBookToBookStore_BOOK_STORE_NAME()).nullsFirst(),
        t2.field(TBookToBookStore_STOCK()).nullsFirst(),
        t2.field(TBookToBookStore_BOOK_ID()))
    .fetch();
```

Renders this SQL:

```sql
select "t2".*
from ("public"."t_book"
  left outer join ("public"."t_book_to_book_store"
    join "public"."t_book_store"
      on "public"."t_book_to_book_store"."book_store_name" = "public"."t_book_store"."name") as "t1"
    on "public"."t_book"."id" = "t1"."book_id") as "t2"
order by
  "t2"."book_store_name" nulls first,
  "t2"."stock" nulls first,
  "t2"."book_id"
```

It would be nicer if the generated SQL was better formatted:$

```sql
select "t2".*
from (
  "public"."t_book"
    left outer join (
      "public"."t_book_to_book_store"
        join "public"."t_book_store"
          on "public"."t_book_to_book_store"."book_store_name" = "public"."t_book_store"."name"
    ) as "t1"
      on "public"."t_book"."id" = "t1"."book_id"
) as "t2"
order by
  "t2"."book_store_name" nulls first,
  "t2"."stock" nulls first,
  "t2"."book_id"
```

**Comment by lukaseder** [comment]
This is really about the aliasing of `JoinTable`, see #14310. Without aliasing, the generated SQL looks well formatted:

```sql
select
  "public"."t_book".*,
  "public"."t_book_to_book_store".*,
  "public"."t_book_store".*
from "public"."t_book"
  left outer join (
    "public"."t_book_to_book_store"
      join "public"."t_book_store"
        on "public"."t_book_to_book_store"."book_store_name" = "public"."t_book_store"."name"
  )
    on "public"."t_book"."id" = "public"."t_book_to_book_store"."book_id"
order by
  "public"."t_book_to_book_store"."book_store_name" nulls first,
  "public"."t_book_to_book_store"."stock" nulls first,
  "public"."t_book_to_book_store"."book_id"
```

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.18.0. Like https://github.com/jOOQ/jOOQ/issues/14310, this won't be backported, due to the various recent changes made to the `QOM` API
